### DISCUSSION TITLE
Mixing `.partial` and  `.default` to add defualt values to each object property

### DISCUSSION THREAD

**Discussion by zomars** [colinhacks/zod#1509] [Q&A]
I have [this playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgLzgXzgMyhEcBEyEAJvgNwBQVA9NXACoAWApgM7NysDGLIAhqzh8oHPgFcYEALQBzZgDtmUPjGbEqXCPNbwAssxAAjJa0bAwAJQgAbDgF5EFOM7i6AoroBCbiwC44AOTuXj4BADROLgCCACK6AJIAcv4BsQmJ4ZHOAPIA6ok+KXkFFgEUaJQUmtrwAPr6RiZmYLokzNZwDsgAdBCGAFbMXDAAFAhZcKp8IPHE-j3yYo1QIwCU3cDyo6sRLnBi7FCz892Ly2sbW2u7LnxcXMxgqnMo3YYQNsx88tcTOLYneQqYAAN2YbjOIwaxigpnMVlsOwmxGArD4hls8XAJi0wK0J3en2+vzQqxodHiATwMCgAE9NjJJhA4PwANYcLTWWmTL4zYhCeT8g5KWZCQQiACOYmAInUFGYAA9ILA4NUdCyDDC4WAAMIiFTMTwkWkAZR4Bj4nTg9U1TXMrWI7W6YGEMGAfGsYwmdweTzU-hpYmYN2c-2YAagQZDcBRaIxzCxYBxQLd+MmkeD5XWjsw4mso3GeymfP8UgAjNHhUcXuXoz7Hs9-LnrOxo2H-Phgt4LPho7H0ZjsbDcan5E2Pa2s5U1fBiCpLQ4QLbYc09V9VEbiKbzfxncJ2GMedNjnAACxhfaHE+n9CrSi0PZwAB6AH4KA+mKJ3mDJrSk5xGAgMRrGIeQAngRg+B-b5uTEQVmEwTY1DgEEPSDVhujgKI8HiFlUVYBlOFwZgYDMeQZDfIA)

I'm trying to mix `.partial` with `.default` but I think the default only works when the parse value is undefined. So how it would be the best way to add a default value to each individual property?

Note that I cannot modify the original schema as it is auto-generated.

EDIT: Added resulting type
<img width="623" alt="imagen" src="https://user-images.githubusercontent.com/3504472/197028436-65b413e9-43cb-4048-919e-6e09c56d9b84.png">

**Comment by scotttrinh** [comment]
You can use `extend` to override a type:

```ts
export const membershipCreateBodySchema = _MembershipModel.extend({
    accepted: _MembershipModel.shape.accepted.default(false),
    role: _MembershipModel.shape.role.default("MEMBER"),
    disableImpersonation: _MembershipModel.shape.disabledImpersonation.default(false),
});
```

[TypeScript Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgLzgXzgMyhEcBEyEAJvgNwBQVA9NXACoAWApgM7NysDGLIAhqzh8oHPgFcYEALQBzZgDtmUPjGbEqXCPNbwAssxAAjJa0bAwAJQgAbDgF5EFOM7i6AoroBCbiwC44AOTuXj4BADROLgCCACK6AJIAcv4BsQmJ4ZHOAPIA6ok+KXkFFgEUaJQUmtrwAPr6RiZmYLokzNZwDsgAdBCGAFbMXDAAFAhZcKp8IPHE-j3yYo1QIwCU3cDyo6sRLnBi7FCz892Ly2sbW2u7LnxcXMxgqnMo3YYQNsx88tcTOLYneQqYAAN2YbjOIwaxigpnMVlsOwmxGArD4hls8XAJi0wK0J3en2+vzQqxodHiATwMCgAE9NjJJhA4PwANYcLTWWmTL4zYhCeT8g5KWZCQQiACOYmAInUFGYAA9ILA4NUdCyDDC4WAAMIiFTMTwkWkAZR4Bj4nTg9U1TXMrWI7W6itUgrGEzuDyean8NuW2odTtMfDAzG6nsez26jsw4mso1j1nYSL2-2YvuhdpabWs3WDoe6aejzFjYnjI3wwW8FnwKZcKLRGOYWNDsNxMGA+OtmdhzUDufzYYb6Mx2LbQI7WmLpfLieTEVJlRgtNDrltvfMeq+qiNxFp8XkYAkVp6myPMAAPMvQxBMBr-c0twbd6bzfwAHyUWh7AB6AH4qGvDge21J8d2NbIJHPE9eigiQrxXZhb3vLVH31cC9zNXg+E-Chv2cf8NC0dViBUS0HBAddQPQw1jSwi1ujAYR2DGHlpmOOAABYwn2Q4OM49BVi-OhfwA78mFEd4wUmRDOEYCAy2IeQAngRg+Gk75uTEQUS02NQ4BBPhrDENhujgKI8HiFlUVYBlOFwZgYDMeQZAAoA)
